### DISCUSSION TITLE
No tips showing in `PlotDataItem`

### DISCUSSION THREAD

**Discussion by noonchen** [pyqtgraph/pyqtgraph#2538] [Q&A] [answered]
Hi, 

I'm using pyqtGraph 0.13.1, I tried to add a `PlotDataItem` to `PlotItem` like the code snippet below, but no tip is showing when I hover on a scatter point, I read the source code and didn't find any code that reads `hoverable` and `tip`.
```python
pitem = pg.PlotItem(viewBox=view)
pitem.addItem(pg.PlotDataItem(x=x, y=y, pen=None,
                                              symbol="o", symbolPen="k",
                                              symbolSize=8, symbolBrush="#00ff00",
                                              hoverable=True, tip='x: {x:.3g}\ny: {y:.3g}\ndata={data}'.format))
```

However, when I add a `ScatterPlotItem` with same arguments to `PlotItem`, hover works but the tip is all white with no text in it, and there is error says:
```python
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/pyqtgraph/graphicsItems/PlotItem/PlotItem.py", line 965, in updateDownsampling
    c.setDownsampling(ds, auto, method)
AttributeError: 'ScatterPlotItem' object has no attribute 'setDownsampling'
```

Any suggestions?

**Comment by noonchen** [comment, marked as answer]
After reading the source code, I manually updated `opts` of inner scatter plot, I don't feel right about this but hover is really working now.
However, the tip is still all white, I cannot see any contents, any help?
<img width="174" alt="image" src="https://user-images.githubusercontent.com/26359487/204087980-ea347737-f652-491d-ac0b-a329ed67bfe4.png">


```python
pdi = pg.PlotDataItem(x=x, y=y, pen=None, 
                                      symbol=fsymbol, symbolPen="k", 
                                      symbolSize=8, symbolBrush=siteColor, 
                                      name=f"Site {site}")
pdi.scatter.opts.update(hoverable=True, 
                                        tip='x: {x:.3g}\ny: {y:.3g}\ndata={data}'.format,
                                        hoverSymbol="+",
                                        hoverSize=12,
                                        hoverPen=pg.mkPen("#ff0000", width=1),
                                        hoverBrush=None)
pitem.addItem(pdi)
```

**Reply by noonchen** [reply]
The issue has been found, the color of the text is set to white, it can be solved by adding a tooltip stylesheet to `GraphicsView`.

**Comment by pijyoi** [comment]
The following seems to work for me.
```python
import pyqtgraph as pg

pg.mkQApp()
pw = pg.PlotWidget()
pw.show()
spi = pg.ScatterPlotItem(hoverable=True, hoverSymbol='+')
pw.addItem(spi)

spi.setData(x=[1,2,3,4], y=[4,3,2,1], data=['A','B','C','D'])

pg.exec()
```

**Reply by noonchen** [reply]
Hi @pijyoi , thanks for your reply.

ScatterPlotItem works for me as well, but I'd like to put it into `pg.PlotItem` instead of `pg.PlotWidget`, because I need to customize `GraphicsView`.

I wonder are there any considerations that exclude hover related arguments in `pg.PlotDataItem`? modify `opts` of inner scatter plot item inside of `pg.PlotDataItem` doesn't look very elegant...

**Reply by pijyoi** [reply]
A call to `PlotWidget::addItem` merely forwards it to the underlying `PlotItem::addItem`
```python
import pyqtgraph as pg

pg.mkQApp()
pw = pg.PlotWidget()
pw.show()
pltitem = pw.getPlotItem()
spi = pg.ScatterPlotItem(hoverable=True, hoverSymbol='+')
pltitem.addItem(spi)

spi.setData(x=[1,2,3,4], y=[4,3,2,1], data=['A','B','C','D'])

pg.exec()
```

**Reply by noonchen** [reply]
The reason I use `PlotDataItem` over `pg.ScatterPlotItem` is because I'd like this method `clipToViewMode`, and the issues that I were facing is all about `PlotDataItem`.

But since I have found the workaround, I don't need to stick to the code elegancy.

thanks @pijyoi
